### ISSUE TITLE
Failed prop type in jest testing

### ISSUE THREAD

**Issue by amanolar94** [the-road-to-learn-react/the-road-to-learn-react#86]
While running the snapshot tests I get the  following warning: The prop 'onClick' is marked as required in Button but its value is undefined.

**Comment by rwieruch** [comment]
Thanks for reporting @amanolar94 I or maybe someone else will check it! :)

**Comment by amanolar94** [comment]
@rwieruch Could it be that this happens because in the tests that we write we don't put an onClick function in the test components? In Table test for example we have some props defined but in Button test not.

EDIT:
I ultimately created a test function and used afterwards and it throws no warnings. What do you think?

```
describe('Button', () => {
	const testFunction = () => true;
	it('renders', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Button onClick={testFunction}>Give Me More</Button>, div);
	});
	test('snapshots', () => {
		const component = renderer.create(
			<Button onClick={testFunction}>Give Me More</Button>
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});`

```

**Comment by rwieruch** [comment]
Afterward it worked, right? 

I think when using and **keeping** prop-types throughout the remaining chapters, there will be a couple of these warnings along the way of forgetting to define a prop. That's why the book mentions in the beginning of the prop-types chapter that all the changes are omitted for the remaining chapters. See https://github.com/the-road-to-learn-react/the-road-to-learn-react/issues/53 Would you think there is some explanation missing around this topic/issue in the book?

**Comment by amanolar94** [comment]
Yes afterwards it worked fine, no warnings. Do you think this solution above is proper?

You are right #53 pretty muchs covers it :P . 
 I wouldn't say there is a problem. It's just that I've been struggling with this for some time and after I saw the fake props in the test code for Table component bellow  I realised that I had to do something similar with the functions also.
I believe most people passed these warnings but if someone wants to get rid of them is mabe gonna have a hard time.
A suggestion would be to put the code for the Table component first so that the reader gets a hint about that or just mention that he has to do that or something similar in order to get rid of these warnings.

Thanks for your time anyway!

**Comment by rwieruch** [comment]
Yes, I think it would be a good step to add the code to fix the first warning coming up after the PropTypes chapter. In your case, was it the first occasion that the warning happened? Then I would add it at this place.

**Comment by amanolar94** [comment]
Well actually I created the PropTypes for Search component also as you suggested so this was the first occasion and the <button> afterwards.

**Comment by lex111** [comment]
@rwieruch can it be closed?

**Comment by rwieruch** [comment]
@lex111 I would want to keep it open as "won't fix", because many people stumble upon this issue while reading the book and they wouldn't necessarily look into the closed issues.

**Comment by teemopower** [comment]
sd
